xds/googledirectpath/googlec2p.go

```diff
@@ -101,14 +101,21 @@ func (c2pResolverBuilder) Build(t resolver.Target, cc resolver.ClientConn, opts
 	if balancerName == "" {
 		balancerName = tdURL
 	}
+	serverConfig := &bootstrap.ServerConfig{
+		ServerURI:    balancerName,
+		Creds:        grpc.WithCredentialsBundle(google.NewDefaultCredentials()),
+		TransportAPI: version.TransportV3,
+		NodeProto:    newNode(<-zoneCh, <-ipv6CapableCh),
+	}
 	config := &bootstrap.Config{
-		XDSServer: &bootstrap.ServerConfig{
-			ServerURI:    balancerName,
-			Creds:        grpc.WithCredentialsBundle(google.NewDefaultCredentials()),
-			TransportAPI: version.TransportV3,
-			NodeProto:    newNode(<-zoneCh, <-ipv6CapableCh),
-		},
+		XDSServer: serverConfig,
 		ClientDefaultListenerResourceNameTemplate: "%s",
+		Authorities: {
+			"traffic-director-c2p.xds.googleapis.com": &bootstrap.Authority{
+				ClientListenerResourceNameTemplate: "%s",
```

Comment:
Does this actually work?

The comment on this field says:
```
	// Must start with "xdstp://<authority_name>/".  If it does not,
	// that is considered a bootstrap file parsing error.
	//
	// If not present in the bootstrap file, defaults to
	// "xdstp://<authority_name>/envoy.config.listener.v3.Listener/%s".
```

Reply:
I commented in https://github.com/grpc/grpc-go/pull/5680#issuecomment-1283073733, but I've tested manually that this works in an end to end setup.

Reply:
This should be resolved one way or another.  If it's supposed to be a parsing error, then it's a bug that we're not throwing a parsing error.  If it's not, then we should update the comment.

Reply:
Ah, my bad.

I think my code was wrong, and the only reason it worked was because in my end-to-end test, we are not actually requesting an initial xdstp LDS resource. I think my code would break when we change the c2p resolver to request xdstp LDS names (e.g. like https://github.com/grpc/grpc/pull/29922/files).

If I'm reading correctly, looks like this field is [only used if we're requesting an xdstp LDS resource](https://github.com/grpc/grpc-go/blob/9127159caf5a3879dad56b795938fde3bc0a7eaa/xds/internal/resolver/xds_resolver.go#L121).

For example, no matter what I set this field to right now, it worked.

So I've reverted this setting and left the field to its default.